File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int cnt = 0, el;
        int ans;
        
        for(auto it: nums){
            if(cnt == 0) el = it;
            
            el == it ? cnt++ : cnt--;
        }
        return el;
    }
};

File: 0169-majority-element/0169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let el=0;
    let cnt=0;
    
    for(const n of nums){
        if(cnt === 0) el = n;
        
        if(el == n) cnt++;
        else cnt--;
    }
    return el;
};